### PR TITLE
Monitor sync committee 

### PR DESCRIPTION
**Pull request by potuz**
This is the sixth PR for the validator monitor. It includes logs and metrics for sync committee contributions. It should only be reviewed after #9910 is merged

### CODE REVIEW

File: beacon-chain/monitor/process_sync_committee.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+			}
+
+			balance, err := state.BalanceAtIndex(validatorIdx)
+			if err != nil {
+				log.Error("Could not get balance")
+				return
+			}
+
+			latestPerf := s.latestPerformance[validatorIdx]
+			balanceChg := balance - latestPerf.balance
```
Comment (rauljordan) [inline beacon-chain/monitor/process_sync_committee.go:54]: Maybe we can use safe sub here

File: beacon-chain/monitor/process_sync_committee.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+			latestPerf.balanceChange = balanceChg
+			latestPerf.balance = balance
+			s.latestPerformance[validatorIdx] = latestPerf
```
Comment (rauljordan) [inline beacon-chain/monitor/process_sync_committee.go:57]: This seems like a shared resource. Could this map ever be updated concurrently? If so we would need a lock

File: beacon-chain/monitor/process_sync_committee.go (hunk 1 above)
Comment (potuz) [reply beacon-chain/monitor/process_sync_committee.go:54]: This surfaced the issue in that `balanceChg` should be `int64` and not `uint64`. There no problem with overflowing here, but indeed the report was bogus. Thanks!

File: beacon-chain/monitor/process_sync_committee.go (hunk 2 above)
Comment (potuz) [reply beacon-chain/monitor/process_sync_committee.go:57]: I was going to study sync map, but decided to go with Locks everywhere, let me know what you think, there's an unexported helper function that assumes a lock inside.

**Review by rauljordan — APPROVED** [review]

File: beacon-chain/monitor/metrics.go [hunk 3]
```diff
@@ -77,4 +78,16 @@ var (
 			"validator_index",
 		},
 	)
+	// syncCommitteeContributionCounter used to track sync committee
+	// contributions
+	syncCommitteeContributionCounter = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "monitor",
+			Name:      "sync_committee_contributions",
```
Comment (terencechain) [inline beacon-chain/monitor/metrics.go:86]:
```suggestion
			Name:      "sync_committee_contributions_total",
```

File: beacon-chain/monitor/service.go [hunk 4]
```diff
@@ -51,3 +61,23 @@ func (s *Service) TrackedIndex(idx types.ValidatorIndex) bool {
 	_, ok := s.config.TrackedValidators[idx]
```
Comment (terencechain) [inline beacon-chain/monitor/service.go:61]: Do you need locks here?

File: beacon-chain/monitor/service.go [hunk 5]
```diff
@@ -40,9 +47,12 @@ type ValidatorMonitorConfig struct {
 // Service is the main structure that tracks validators and reports logs and
 // metrics of their performances throughout their lifetime.
 type Service struct {
-	config                *ValidatorMonitorConfig
-	latestPerformance     map[types.ValidatorIndex]ValidatorLatestPerformance
-	aggregatedPerformance map[types.ValidatorIndex]ValidatorAggregatedPerformance
+	config                      *ValidatorMonitorConfig
+	monitorLock                 sync.RWMutex
```
Comment (terencechain) [inline beacon-chain/monitor/service.go:51]: it's not clear to me what this lock is protecting, `config`? `latestPerformance`? or everything? Perhaps a comment or better name would suffice

**Review by terencechain — CHANGES_REQUESTED** [review]
Should the lock protect against `s.config.TrackedValidators`?

**Comment by potuz** [comment]
> Should the lock protect against `s.config.TrackedValidators`?

We currently do not allow changes in that config, I thought of adding that after the service is working when we add also support for automatic discovery

**Comment by terencechain** [comment]
> > Should the lock protect against `s.config.TrackedValidators`?
> 
> We currently do not allow changes in that config, I thought of adding that after the service is working when we add also support for automatic discovery

My recommendation is to add it now and just use the same lock for everything to reduce disparity so we dont end up with two locks (hard to implement) or one lock which is same as we implement that today. Less headache later on

File: beacon-chain/monitor/service.go [hunk 6]
```diff
@@ -61,7 +61,7 @@ type Service struct {
 
 // TrackedIndex returns if the given validator index corresponds to one of the
 // validators we follow.
-// It assumes the caller holds a Lock on the monitorLock
+// It assumes the caller holds a Lock on the Lock
```
Comment (terencechain) [inline beacon-chain/monitor/service.go:64]: `holds a Lock on the Lock` sounds funny : )

**Review by terencechain — APPROVED** [review]

**Review by nisdas — APPROVED** [review]

**Pull request merged**